lib/std/net.zig

```diff
@@ -1374,12 +1412,20 @@ pub const StreamServer = struct {
     };
 
     /// If this function succeeds, the returned `Connection` is a caller-managed resource.
-    pub fn accept(self: *StreamServer) AcceptError!Connection {
-        const nonblock = if (std.io.is_async) os.SOCK_NONBLOCK else 0;
-        const accept_flags = nonblock | os.SOCK_CLOEXEC;
+    pub fn accept(self: *StreamServer) !Connection {
         var accepted_addr: Address = undefined;
         var adr_len: os.socklen_t = @sizeOf(Address);
-        if (os.accept4(self.sockfd.?, &accepted_addr.any, &adr_len, accept_flags)) |fd| {
+        var _accept: os.AcceptError!os.fd_t = undefined;
+
+        if(comptime builtin.os.tag.isDarwin()) {
+            try setUnixSocketFlags(self.sockfd.?);
```

Comment:
shouldn't this be below `os.accept`, and operate on `_accept`?

Reply:
On macOS (and other BSDs) the new file descriptor returned by accept() will inherit the same flags as the ones passed into it. Only Linux clears the file descriptor on accept() and you then have to do it again after with fcntl. That is, unless you use accept4 and set the flags in the arguments (like we do). Then the new file descriptor will have those flags as well.  So this should be OK.